src/amo/containers/App.js

```diff
@@ -18,38 +19,48 @@ export class AppBase extends React.Component {
     handleLogIn: PropTypes.func.isRequired,
     i18n: PropTypes.object.isRequired,
     isAuthenticated: PropTypes.bool,
+    lang: PropTypes.string.isRequired,
     location: PropTypes.object.isRequired,
     MastHeadComponent: PropTypes.node,
+    query: PropTypes.string,
+    SearchFormComponent: PropTypes.node,
   }
 
   static defaultProps = {
     MastHeadComponent: MastHead,
+    SearchFormComponent: SearchForm,
   }
 
   accountButton() {
     const { handleLogIn, i18n, isAuthenticated, location } = this.props;
     return (
-      <button className="button AccountButton" onClick={() => handleLogIn(location)}
+      <button className="button AccountButton"
+              onClick={() => handleLogIn(location)}
               ref={(ref) => { this.logInButton = ref; }}>
         <span>{ isAuthenticated ? i18n.gettext('Log out') : i18n.gettext('Log in/Sign up') }</span>
       </button>
     );
   }
 
   render() {
-    const { MastHeadComponent, children, i18n } = this.props;
+    const { MastHeadComponent, SearchFormComponent, children, i18n, lang,
+            query } = this.props;
+    const pathname = `/${lang}/firefox/search/`;
     return (
       <div className="amo">
         <Helmet defaultTitle={i18n.gettext('Add-ons for Firefox')} />
-        <MastHeadComponent>{this.accountButton()}</MastHeadComponent>
+        <header className="App-header">
+          <MastHeadComponent>{this.accountButton()}</MastHeadComponent>
+          <SearchFormComponent pathname={pathname} query={query} />
```

Comment:
I think it might be nicer as child of the MastHead, now you've got `header` inside a `header` which seems a bit odd